### ISSUE TITLE
Incorrect reporting of errors when deleting line breaks

### ISSUE THREAD

**Issue by circumspect** [jeapostrophe/racket-langserver#54]
Suppose I have a file containing the following dummy function:

```scheme
#lang racket

(define (hello n) (cadr (list 6 3 6 2)))
```
If I insert a line break right before `(cadr ...)` (not including the space between n and cadr) to get the following:
```scheme
#lang racket

(define (hello n) 
(cadr (list 6 3 6 2)))
```
and then delete the line break, I receive an error that does not disappear until I close and reopen the file.

After some testing, it seems that every time that I repeat the steps of adding a line break and then deleting it, the error is reported as being at a position one space to the left of where it was previously reported, and after enough repeats, the error type being reported changes. For instance, the example above is reported as the following (each new error message comes after inserting and then deleting the newline at the same position):

`test.rkt:3:38: read-syntax: unexpected ')'`
`test.rkt:3:37: read-syntax: unexpected ')'`
`test.rkt:3:36: read-syntax: unexpected ')'`
...
`test.rkt:3:21: read-syntax: unexpected ')'`
`test.rkt:3:20: read-syntax: unexpected ')'`
`test.rkt:3:19: read-syntax: unexpected ')'`
`test.rkt:3:19: read-syntax: unexpected ')'` (not a typo, it says 19 again)
`test.rkt:3:0: define: bad syntax (no expressions for procedure body`
`test.rkt:3:0: read-syntax: expected a ')' to close '('`
`test.rkt:3:0: read-syntax: expected a ')' to close '('`

Beyond this point it just repeats the last error about expected ')'.

Interestingly, when adding and deleting the line break BEFORE the space (i.e. right after `(hello n)` this error does not occur until the second time that the line break is deleted, or if the file is saved before deleting the line break the first time.

**Comment by jeapostrophe** [comment]
Can you let me know what platform you're on? I suspect that something is inconsistent about how the file gets represented on the LSP end. I've added a `debug?` option in `debug.rkt` and it will create `debug.out.rkt` as it tries to check the files. Can you download the git repo, install it with `raco pkg install`, and then run your test again and see what's in that file?

**Comment by circumspect** [comment]
I installed the latest version from git (with `raco pkg install --clone github.com/jeapostrophe/racket-langserver`, let me know if that's incorrect), and attempted to recreate the bug, but was unsuccessful, so I uninstalled that version and reinstalled the old one (`raco pkg install racket-langserver`), and the bug still won't show up. I'm not sure if there's anything that can be done, so I'm closing this issue unless it reappears.

As for the information you requested:
OS: Windows 10
Editor: VS Code (1.55.2)
Extension: Magic Racket (0.5.7)
I was unable to locate a file named `debug.out.rkt` on my computer. If you would like any more information, let me know
